pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,24 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
```

Comment:
Is ADC locked at this time? I am not sure, but I think the reconciler runs in a separate goroutine and it could run at this point, where we have node and its attached volumes in states of the world, while the pods are not there yet. reconcile() will try detach these volumes.